esmvaltool/namelist.py

```diff
@@ -288,16 +288,16 @@ def _get_default_settings(variable, config_user):
         'project': variable['project'],
         'model': variable['model'],
         'short_name': variable['short_name'],
-        'preproc_dir': config_user['preproc_dir'],
+        'var_path': variable['var_path'],
```

Comment:
You're right that there can be multiple fixed output files, I didn't think about that. How about replacing the line above by a more descriptive:
```
'output_dir': variable['filename'] + '_fixed'
```
and then creating any fixed files in that directory. The same directory can then be provided to the new `cleanup` function, e.g. 
```
settings['cleanup'] = {
    'remove':  variable['filename'] + '_fixed',
}
````
to be deleted at the very end of the preprocessor run.

Reply:
Nice and cleaner

Reply:
If save_intermediate_files is True, we should skip the cleanup, no?